reducers/web_reducer.js

```diff
@@ -13,130 +14,102 @@ const INITIAL_STATE = {
-                ...state,
-                photos: webPhotos
-            };
-
-        /**
-         * Remove a tag that has been pressed
-         */
-        case REMOVE_TAG_FROM_WEB_IMAGE:
-            let updatedWebPhotos = [...state.photos];
-
-            let img = updatedWebPhotos[action.payload.currentIndex];
-            delete img.tags[action.payload.category][action.payload.tag];
-
-            // Delete the category if empty
-            if (Object.keys(img.tags[action.payload.category]).length === 0) {
-                delete img.tags[action.payload.category];
-            }
-
-            return {
-                ...state,
-                photos: updatedWebPhotos
-            };
-
-        /**
-         * At the end of swiping web.photos, load more images
-         */
-        case LOAD_MORE_WEB_IMAGES:
-            let web_images = [...state.photos];
-
-            action.payload.photos.forEach(photo => {
-                web_images.push(photo);
-            });
-
-            return {
-                ...state,
-                photos: web_images
-            };
-
-        /**
-         * After submitting an image + tags, we filter it from the photos array
-         *
-         * and decrement the count
-         */
-        case REMOVE_WEB_IMAGE:
-            const filtered = state.photos.filter(
-                photo => photo.id !== action.payload
-            );
-
-            const count = state.count - 1;
-
-            return {
-                ...state,
-                photos: filtered,
-                count
-            };
-
-        /**
-         * Images have been uploaded from the web
-         */
-        case WEB_IMAGES:
-            const photos = (state.photos = [
-                ...state.photos.slice(0, 0),
-                ...state.photos.slice(1)
-            ]);
-
-            return {
-                ...state,
-                count: action.payload.count,
-                photos: action.payload.photos
-            };
 
-        default:
-            return state;
-    }
+                break;
+
+            /**
+             * At the end of swiping web.photos, load more images
+             */
+            case LOAD_MORE_WEB_IMAGES:
+                action.payload.map(photo => {
+                    draft.photos.push(photo);
+                    draft.count++;
+                });
+                break;
+
+            /**
+             * After submitting an image + tags, we filter it from the photos array
+             *
+             * and decrement the count
+             */
+            case REMOVE_WEB_IMAGE:
+                const index = draft.photos.findIndex(
+                    photo => photo.id === action.payload
+                );
+                if (index !== -1) draft.photos.splice(index, 1);
+
+                draft.count = draft.count - 1;
+
+                break;
+
+            /**
+             * Images have been uploaded from the web
+             */
+            case WEB_IMAGES:
```

Comment:
Reducer name is not clear. GET_WEB_IMAGES?

Reply:
This PR is just refactoring old reducer to use immer.
So I will create a new PR where I will change the reducer name to make it more easy to understand...